corehq/apps/app_manager/models.py

```diff
@@ -4444,23 +4475,32 @@ def generate_shortened_url(self, url_type):
         else:
             return shortened_url
 
-    def get_short_url(self):
-        if not self.short_url:
-            self.short_url = self.generate_shortened_url('download_jad')
-            self.save()
-        return self.short_url
-
-    def get_short_odk_url(self, with_media=False):
-        if with_media:
-            if not self.short_odk_media_url:
-                self.short_odk_media_url = self.generate_shortened_url('download_odk_media_profile')
+    def get_short_url(self, build_profile_id=None):
+        if not build_profile_id:
+            if not self.short_url:
+                self.short_url = self.generate_shortened_url('download_jad')
                 self.save()
-            return self.short_odk_media_url
+            return self.short_url
         else:
-            if not self.short_odk_url:
-                self.short_odk_url = self.generate_shortened_url('download_odk_profile')
-                self.save()
-            return self.short_odk_url
+            return self.generate_shortened_url('download_jad', build_profile_id)
```

Comment:
In all the other cases, when `self.generate_shortened_url(...)` is called, `self.save()` is called immediately afterwards. Is that somehow not needed here?